ptype/full_rap_apply.py

```diff
@@ -0,0 +1,167 @@
+
+    print("Retrieved mPING observations.")
+
+    # Map predictions by p-type
+    bbox = [50.0, 20.0, -63.0, -125.0]
+    proj = ccrs.LambertConformal(central_longitude=(bbox[2] + bbox[3])/2, central_latitude=(bbox[0] + bbox[1])/2)
+    lats, lons = rap_data["latitude"].data, rap_data["longitude"].data-360
+
+    p_levels = [i*0.1 for i in range(11)]
+
+    fig, axarr = plt.subplots(nrows=2, ncols=2, figsize=(18, 12), constrained_layout=True,
+                            subplot_kw={'projection': proj})
+    fig.suptitle(f"mPING Simple MLP on RAP {date} {time} UTC", fontsize=18)
+    axlist = axarr.flatten()
+    for ax in axlist:
+        plot_background(ax, bbox)
+        
+    ra = axlist[0].contourf(lons.T, lats.T, preds[:, :, 0], levels=p_levels, transform=ccrs.PlateCarree(), cmap=plt.get_cmap("Greens"))
+    clb = fig.colorbar(ra, fraction=0.025, pad=0.04, ax=axlist[0])
+    axlist[0].set_title("Rain Probabilities", fontsize=16)
+    axlist[0].scatter(df_time["lon"][df_time["ptype_max"] == 0]-360, df_time["lat"][df_time["ptype_max"] == 0],
+                        c=df_time["ptype_max"][df_time["ptype_max"] == 0].map(colors),
+                        s=50, transform=ccrs.PlateCarree())
+
+    sn = axlist[1].contourf(lons.T, lats.T, preds[:, :, 1], levels=p_levels, transform=ccrs.PlateCarree(), cmap=plt.get_cmap("Blues"))
+    clb = fig.colorbar(sn, fraction=0.025, pad=0.04, ax=axlist[1])
+    axlist[1].set_title("Snow Probabilities", fontsize=16)
+    axlist[1].scatter(df_time["lon"][df_time["ptype_max"] == 1]-360, df_time["lat"][df_time["ptype_max"] == 1],
+                        c=df_time["ptype_max"][df_time["ptype_max"] == 1].map(colors),
+                        s=50, transform=ccrs.PlateCarree())
+
+    pl = axlist[2].contourf(lons.T, lats.T, preds[:, :, 2], levels=p_levels, transform=ccrs.PlateCarree(), cmap=plt.get_cmap("Oranges"))
+    clb = fig.colorbar(pl, fraction=0.025, pad=0.04, ax=axlist[2])
+    axlist[2].set_title("Ice Pellet Probabilities", fontsize=16)
+    axlist[2].scatter(df_time["lon"][df_time["ptype_max"] == 2]-360, df_time["lat"][df_time["ptype_max"] == 2],
+                        c=df_time["ptype_max"][df_time["ptype_max"] == 2].map(colors),
+                        s=50, transform=ccrs.PlateCarree())
+
+    fzra = axlist[3].contourf(lons.T, lats.T, preds[:, :, 3], levels=p_levels, transform=ccrs.PlateCarree(), cmap=plt.get_cmap("Purples"))
+    clb = fig.colorbar(fzra, fraction=0.025, pad=0.04, ax=axlist[3])
+    axlist[3].set_title("Freezing Rain Probabilities", fontsize=16)
+    axlist[3].scatter(df_time["lon"][df_time["ptype_max"] == 3]-360, df_time["lat"][df_time["ptype_max"] == 3],
+                        c=df_time["ptype_max"][df_time["ptype_max"] == 3].map(colors),
+                        s=50, transform=ccrs.PlateCarree())
+
+    plt.savefig(map_save_loc + f"/full_rap_pred_{date}_{time}_ptypes.jpg", dpi=300)
```

Comment:
Make the output file format a keyword argument and use "png" as the default rather than "jpg." Add the argument `bbox_inches="tight"` to savefig to reduce the whitespace around the edge of the image.